src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EgressExtensibilityTests.cs

```diff
@@ -0,0 +1,214 @@
+
+        private IEgressExtension FindEgressExtension(ConfigDirectory configDirectory, string exePath = null)
+        {
+            HostBuilderSettings settings = CreateHostBuilderSettings();
+
+            string directoryName = GetExtensionDirectoryName(settings, configDirectory);
+
+            string extensionDirPath = configDirectory == ConfigDirectory.DotnetToolsExtensionDirectory ? Path.Combine(directoryName, DotnetToolsExtensionDir) : Path.Combine(directoryName, ExtensionsFolder, AppName);
+
+            CopyExtensionFiles(extensionDirPath, exePath);
+
+            IHost host = TestHostHelper.CreateHost(_outputHelper, rootOptions => { }, host => { }, settings: settings);
+
+            var extensionDiscoverer = host.Services.GetService<ExtensionDiscoverer>();
+
+            return extensionDiscoverer.FindExtension<IEgressExtension>(ProviderName);
+        }
+
+        private static async Task GetStream(Stream stream, CancellationToken cancellationToken)
+        {
+            // The test extension currently does not do anything with this stream.
+            await stream.WriteAsync(ByteArray, cancellationToken);
+        }
+
+        private static void CopyExtensionFiles(string extensionDirPath, string exePath = null)
+        {
+            Directory.CreateDirectory(extensionDirPath);
+
+            string testAppDirPath = Path.GetDirectoryName(AssemblyHelper.GetAssemblyArtifactBinPath(Assembly.GetExecutingAssembly(), AppName));
+
+            bool hasSeparateExe = !string.IsNullOrEmpty(exePath);
+
+            foreach (string testAppFilePath in Directory.GetFiles(testAppDirPath, "*.*", SearchOption.AllDirectories))
+            {
+                string extensionFilePath;
+
+                if (hasSeparateExe && IsExecutablePath(testAppFilePath))
+                {
+                    Directory.CreateDirectory(exePath);
+                    extensionFilePath = testAppFilePath.Replace(testAppDirPath, exePath);
+                }
+                else
+                {
+                    extensionFilePath = testAppFilePath.Replace(testAppDirPath, extensionDirPath);
+                }
+
+                Directory.CreateDirectory(Path.GetDirectoryName(extensionFilePath));
+
+                File.Copy(testAppFilePath, extensionFilePath, true);
+            }
+        }
+
+        private static bool IsExecutablePath(string path)
+        {
+            if (Path.GetFileNameWithoutExtension(path) == AppName)
+            {
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows) && Path.GetExtension(path) == ".exe")
+                {
+                    return true;
+                }
+                else
+                {
+                    return Path.GetExtension(path) == string.Empty;
+                }
+            }
+
+            return false;
+        }
+
+        private HostBuilderSettings CreateHostBuilderSettings()
+        {
+            TemporaryDirectory executingAssemblyDir = new(_outputHelper);
```

Comment:
All of these are disposable, so you are leaking temporary directories for each invocation of this method.